include/inviwo/core/properties/listproperty.h

```diff
@@ -141,13 +141,13 @@ class IVW_CORE_API ListProperty : public CompositeProperty {
     void clear();
 
     /**
-     * \brief add a list entry which is created from the respective prefab object.
+     * \brief construct a list entry which is created from the respective prefab object.
      * This function has no effect if the list size will exceed the maximum number of elements.
      *
      * @param prefabIndex   index of prefab object used for creating the new entry
      * @throw RangeException  in case prefabIndex is invalid
      */
-    void addProperty(size_t prefabIndex);
+    Property* constructProperty(size_t prefabIndex);
```

Comment:
Is there a special reason for renaming the function? 
Implementation has a code path for returning null. Would be good to mention if that can occur in the documentation (not sure how that differs from range exception).

Reply:
Yes, a very specific reason, i.e. function overload. This refactoring was done primarily to prevent future problems e.g. feature creep!
Before, you could not call `ListProperty::addProperty(0);` without explicitly casting the `0` to `size_t` as it is ambiguous.

Reply:
The range exception is tied to the prefab index while not being able to add new items (due to max size) produces a log error and returns `nullptr`. The latter being considered not as severe.